package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
@lucboudreau this is JavaScript and we don't have the fortune of an assert statement :-(